client/blocks/image-editor/index.jsx

```diff
@@ -88,84 +89,30 @@ const ImageEditor = React.createClass( {
 
 	onDone() {
 		const canvasComponent = this.refs.editCanvas.getWrappedInstance();
-		canvasComponent.toBlob( this.onImageExtracted );
 
-		if ( this.props.onImageEditorSave ) {
-			this.props.onImageEditorSave();
-		}
-	},
-
-	onCancel() {
-		this.props.resetAllImageEditorState();
-
-		if ( this.props.onImageEditorCancel ) {
-			this.props.onImageEditorCancel();
-		}
-	},
-
-	onImageExtracted( blob ) {
-		const {
-			fileName,
-			site,
-			translate
-		} = this.props;
-
-		const mimeType = MediaUtils.getMimeType( fileName );
-
-		// check if a title is already post-fixed with '(edited copy)'
-		const editedCopyText = translate(
-			'%(title)s (edited copy)', {
-				args: {
-					title: ''
-				}
-			} );
-
-		let { title } = this.props;
-
-		if ( title.indexOf( editedCopyText ) === -1 ) {
-			title = translate(
-				'%(title)s (edited copy)', {
-					args: {
-						title: title
-					}
-				} );
-		}
-
-		this.props.resetAllImageEditorState();
+		canvasComponent.toBlob( ( blob ) => {
```

Comment:
I see. But why would we want to avoid that? I actually like that it's on one place and I don't have to jump to the `this.onImageExtracted` method to see what it does. :)